datafusion/tests/sql/joins.rs

```diff
@@ -210,6 +210,26 @@ async fn left_join_unbalanced() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn left_join_filter_pushdown() -> Result<()> {
+    // Since t2 is the non-preserved side of the join, we cannot push down a NULL filter.
+    let mut ctx = create_join_context_with_nulls()?;
+    let sql = "SELECT t1_id, t2_id, t2_name FROM t1 LEFT JOIN t2 ON t1_id = t2_id WHERE t2_name IS NULL ORDER BY t1_id";
```

Comment:
👍  this does match postgres:

```sql
alamb=# create table t1(t1_id, t1_name) as select * from (values (11, 'a'), (22, 'b'), (33, 'c'), (44, 'd'), (77, 'e')) as sq;
SELECT 5
alamb=# create table t2(t2_id, t2_name) as select * from (values (11, 'z'), (22, null), (44, 'x'), (55, 'w')) as sq;
SELECT 4
alamb=# SELECT t1_id, t2_id, t2_name FROM t1 LEFT JOIN t2 ON t1_id = t2_id WHERE t2_name IS NULL ORDER BY t1_id;
 t1_id | t2_id | t2_name 
-------+-------+---------
    22 |    22 | 
    33 |       | 
    77 |       | 
(3 rows)

```


I think the key would be to test adding a filter on `t2_id` however (as that is the predicate type that is created)